libtiledbsoma/src/soma/soma_array.cc

```diff
@@ -169,11 +169,11 @@ SOMAArray::SOMAArray(
     std::optional<TimestampRange> timestamp)
     : uri_(util::rstrip_uri(arr->uri()))
     , ctx_(ctx)
+    , arr_(arr)
```

Comment:
Please add a comment here about how the initializer order is important, & controls the order that destructors run.

I know this is "general C++ knowledge" but this is (was!) a _particular_ pain-point and it deserves its own "hazard warning" sign.